src/Microsoft.ML.StaticPipe/SdcaStaticExtensions.cs

```diff
@@ -115,13 +150,69 @@ public static class SdcaStaticExtensions
+                    if (onFit != null)
+                    {
+                        return trainer.WithOnFitDelegate(trans =>
+                        {
+                            // Under the default log-loss we assume a calibrated predictor.
+                            var model = trans.Model;
+                            var cali = (ParameterMixingCalibratedPredictor)model;
+                            var pred = (LinearBinaryModelParameters)cali.SubPredictor;
+                            onFit(pred, cali);
+                        });
+                    }
+                    return trainer;
+                }, label, features, weights);
+
+            return rec.Output;
+        }
+
+        /// <summary>
+        /// Predict a target using a linear binary classification model trained with the SDCA trainer, and log-loss.
+        /// </summary>
+        /// <param name="ctx">The binary classification context trainer object.</param>
+        /// <param name="label">The label, or dependent variable.</param>
+        /// <param name="features">The features, or independent variables.</param>
+        /// <param name="weights">The optional example weights.</param>
+        /// <param name="options">Advanced arguments to the algorithm.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained, as well as the calibrator on top of that model. Note that this action cannot change the
+        /// result in any way; it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The set of output columns including in order the predicted binary classification score (which will range
+        /// from negative to positive infinity), the calibrated prediction (from 0 to 1), and the predicted label.</returns>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[SDCA](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Static/SDCABinaryClassification.cs)]
+        /// ]]></format>
+        /// </example>
+        public static (Scalar<float> score, Scalar<float> probability, Scalar<bool> predictedLabel) Sdca(
+                    this BinaryClassificationContext.BinaryClassificationTrainers ctx,
+                    Scalar<bool> label, Vector<float> features, Scalar<float> weights,
+                    SdcaBinaryTrainer.Options options,
+                    Action<LinearBinaryModelParameters, ParameterMixingCalibratedPredictor> onFit = null)
+        {
+            Contracts.CheckValue(label, nameof(label));
```

Comment:
Just curious - this is a documented example right? Is it worth having CheckValues in the example?

Reply:
this is the Static API extension

---
In reply to: [248873167](https://github.com/dotnet/machinelearning/pull/2093#discussion_r248873167) [](ancestors = 248873167)